src/amo/reducers/userAbuseReports.js

```diff
@@ -0,0 +1,251 @@
+/* @flow */
+import type { UserType } from 'amo/reducers/users';
+
+export const ABORT_USER_ABUSE_REPORT: 'ABORT_USER_ABUSE_REPORT' =
+  'ABORT_USER_ABUSE_REPORT';
+export const HIDE_USER_ABUSE_REPORT_UI: 'HIDE_USER_ABUSE_REPORT_UI' =
+  'HIDE_USER_ABUSE_REPORT_UI';
+export const LOAD_USER_ABUSE_REPORT: 'LOAD_USER_ABUSE_REPORT' =
+  'LOAD_USER_ABUSE_REPORT';
+export const SEND_USER_ABUSE_REPORT: 'SEND_USER_ABUSE_REPORT' =
+  'SEND_USER_ABUSE_REPORT';
+export const SHOW_USER_ABUSE_REPORT_UI: 'SHOW_USER_ABUSE_REPORT_UI' =
+  'SHOW_USER_ABUSE_REPORT_UI';
+
+
+type AbortUserAbuseReportParams = {|
+  user: UserType,
```

Comment:
I explained before but passing around the User object when we already have a `UserType` seemed nicer than making a new type–which is just a number so arguably is less of a strict type. There doesn't seem to be a downside to passing around the entire object (and we aren't storing the whole user object again in state), and this way we can use more info later if we like. Also I just liked passing around `user` over `userId: user.id`. I don't see it simplifying anything in terms of the complexity of the code. Also, it arguably makes debugging a bit easier as you can see more about the user when using the Redux devtools and the action's payload.

If there's some performance hit to passing the entire `User` object around (which, if anything, there shouldn't be as it should be passed as a reference in JS rather than as a new integer when we pass just the ID) then I think it makes sense to change, but otherwise it seems to be just an aesthetic choice.